yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -66,6 +66,9 @@ private[spark] class Client(
   import Client._
   import YarnSparkHadoopUtil._
 
+  val context: String = s"${sparkConf.get("spark.app.name")} running on Spark"
+  Utils.setCallerContext(context)
```

Comment:
If spark applications are in Yarn cluster mode, the record with that caller context below will be in HDFS log:
2016-07-21 14:32:33,404 INFO FSNamesystem.audit: allowed=true	ugi=wyang (auth:SIMPLE)	ip=/127.0.0.1	cmd=getfileinfo	src=/spark-history	dst=null	perm=null	proto=rpc	callerContext=org.apache.spark.examples.SparkKMeans running on Spark